core/chains/evm/client/node_lifecycle.go

```diff
@@ -173,6 +173,9 @@ func (n *node) aliveLoop() {
 				lggr.Tracew("Ignoring previously seen block number", "latestReceivedBlockNumber", latestReceivedBlockNumber, "blockNumber", bh.Number, "nodeState", n.State())
 			}
 			outOfSyncT.Reset(noNewHeadsTimeoutThreshold)
+			n.stateMu.Lock()
+			n.latestReceivedBlockNumber = bh.Number
```

Comment:
Can we drop the local `latestReceivedBlockNumber` var too?

Reply:
Looks like we cannot drop it, it breaks many tests, because this var is tracking the historical highest number, whereas the struct member tracks whatever number we received.

Reply:
That seems problematic and possibly an oversight/bug. @samsondav 
Should we at least rename it then?

Reply:
When I start working on the tests, I can verify all node_lifecycle_test tests and fix if I find a bug in test.

Reply:
Yeah it probably ought to move entirely onto the node and not need its own local var in tests.

Reply:
Is there a difference between latestReceived and highestReceived?